m4opt/missions/_lsst/camera.py

```diff
@@ -0,0 +1,115 @@
+    - Offsets and pixel sizes are provided in millimeters (mm).
+    - The data is parsed directly from YAML into an Astropy Table and used to make the FOV.
+"""
+
+from importlib import resources
+
+import yaml
+from astropy import units as u
+from astropy.coordinates import SkyCoord
+from astropy.table import Table
+from regions import PolygonSkyRegion, RectangleSkyRegion, Regions
+
+from . import data
+
+
+class LSSTCameraFOV:
+    """LSST Field of View (FOV) based on detector positions."""
+
+    PLATE_SCALE = 0.2 * u.arcsec  # arcsec/pixel
+    PIXEL_SIZE = 0.01  # mm/pixel
+    MM_TO_ARCSEC = PLATE_SCALE / PIXEL_SIZE
+
+    # LSST Detector Bounding Box Sizes (Pixels)
+    LSST_DETECTOR_SIZES = {
+        "E2V": [4095, 4003],  # LSST detectors from E2V
+        "ITL": [4071, 3999],  # LSST detectors from ITL
+    }
+
+    def __init__(self):
+        """Initialize LSST camera with detector data loaded from YAML."""
+        self.detectors = self.read_yaml()
+
+    def read_yaml(self):
+        """
+        Read LSST science detectors from YAML (excluding wavefront and guide sensors).
+        """
+        file_path = resources.files(data) / "lsstCamSim.yaml"
+        with open(file_path, "r") as file:
+            camera_data = yaml.safe_load(file)
+
+        science_detectors = []
+        for det_name, det_info in camera_data.get("CCDs", {}).items():
+            detector_type = det_info.get("detectorType")
+
+            if detector_type != 0 or any(sub in det_name for sub in ["SG", "SW"]):
+                continue
+
+            detector_offset = det_info.get("offset", [0, 0])
+            physical_type = det_info.get("physicalType", "Unknown")
+
+            science_detectors.append(
+                {
+                    "detector_name": det_name,
+                    "x_offset": detector_offset[0],
+                    "y_offset": detector_offset[1],
+                    "physical_type": physical_type,
+                }
+            )
+
+        table = Table(
+            rows=science_detectors,
+            names=["detector_name", "x_offset", "y_offset", "physical_type"],
+        )
+        return table
+
+    @staticmethod
+    def get_bbox_size(physical_type):
+        """Return detector bounding box size (width, height) in pixels."""
+        return LSSTCameraFOV.LSST_DETECTOR_SIZES.get(physical_type, None)
+
+    @staticmethod
+    def convert_rectangle_to_polygon_skyregion(rect):
+        """Convert RectangleSkyRegion to PolygonSkyRegion in RA/Dec."""
+        ra, dec = rect.center.ra.deg, rect.center.dec.deg
+        half_w = rect.width.to_value(u.deg) / 2
+        half_h = rect.height.to_value(u.deg) / 2
+        corners_ra = [ra - half_w, ra + half_w, ra + half_w, ra - half_w]
+        corners_dec = [dec - half_h, dec - half_h, dec + half_h, dec + half_h]
+        return PolygonSkyRegion(
+            vertices=SkyCoord(ra=corners_ra * u.deg, dec=corners_dec * u.deg)
+        )
```

Comment:
I don't see why you need this method. What is wrong with leaving the regions as RectangleSkyRegion instances?